### PR TITLE
Fix uninitialized min_rand variable in Thompson MP when using SPP

### PR DESCRIPTION
**Pull request by JeffBeck-NOAA**
## Description

This suite of PRs fixes the uninitialized min_rand variable in Thompson MP when using SPP and is explained in detail in the ufs-weather-model PR [here](https://github.com/ufs-community/ufs-weather-model/pull/1152).

### Issue(s) addressed

Fixes NCAR/ccpp-physics [Issue #891](https://github.com/NCAR/ccpp-physics/issues/891).

## Testing

The RT suite on Hera passed for all normal and debug tests, except for the regional_spp_sppt_shum_skeb RT test, which will now have a new baseline result.

## Dependencies

Depends on the following PR from [ccpp-physics](https://github.com/NCAR/ccpp-physics/pull/892).

@jwolff-ncar, @willmayfield, @bluefinweiwei, @michelleharrold, @judithberner, @gsketefian

### CODE REVIEW

**Review by SamuelTrahanNOAA — APPROVED** [review]

**Review by SamuelTrahanNOAA — CHANGES_REQUESTED** [review]
This PR will not have consistent results between thread counts due to this line in module_mp_thompson.F90

```
abs_min_rand = ABS(MINVAL(rand_pert(:,1)))
```

**Comment by JeffBeck-NOAA** [comment]
@SamuelTrahanNOAA, please see the latest changes to this PR, which I believe will resolve your concerns. The code is now working (ran the RT test; baselines change) and calculates the maximum random perturbation value based on namelist settings brought in from stochastic_physics.

**Comment by SamuelTrahanNOAA** [comment]
There is a minor issue in the mp_thompson.meta, but other than that it works at and below the fv3atm level.

**Comment by JeffBeck-NOAA** [comment]
Thanks @SamuelTrahanNOAA.  I fixed the metadata issues in the ccpp/physics repo PR.

**Review by SamuelTrahanNOAA — APPROVED** [review]

File: stochastic_physics/stochastic_physics_wrapper.F90 [hunk 1]
```diff
@@ -109,7 +109,7 @@ subroutine stochastic_physics_wrapper (GFS_Control, GFS_Data, Atm_block, ierr)
             GFS_Control%input_nml_file, GFS_Control%fn_nml, GFS_Control%nlunit, xlon, xlat, GFS_Control%do_sppt, GFS_Control%do_shum, &
             GFS_Control%do_skeb, GFS_Control%lndp_type, GFS_Control%n_var_lndp, GFS_Control%use_zmtnblck, GFS_Control%skeb_npass,     &
             GFS_Control%lndp_var_list, GFS_Control%lndp_prt_list,    &
-            GFS_Control%n_var_spp, GFS_Control%spp_var_list, GFS_Control%spp_prt_list, GFS_Control%do_spp,                            &
+            GFS_Control%n_var_spp, GFS_Control%spp_var_list, GFS_Control%spp_prt_list, GFS_Control%spp_stddev_cutoff, GFS_Control%do_spp,                            &
```
Comment (pjpegion) [inline stochastic_physics/stochastic_physics_wrapper.F90:112]: This has a different order than what is in stochastic_physics/stochastic_physcs.F90
here: 
GFS_Control%spp_var_list, GFS_Control%spp_prt_list, GFS_Control%spp_stddev_cutoff,

in stochastic_physics.F90:
spp_var_list_out, spp_stddev_cutoff_out, spp_prt_list_out

File: ccpp/data/GFS_typedefs.meta [hunk 2]
```diff
@@ -4858,17 +4858,24 @@
...
   type = integer
 [spp_prt_list]
-  standard_name =magnitude_of_spp_perturbations
+  standard_name = magnitude_of_spp_perturbations
   long_name = magnitude of spp perturbations
   units = 1
-  dimensions = (number_of_spp_schemes_perturbed)
+  dimensions = (number_of_perturbed_spp_schemes)
+  type = real
+  kind = kind_phys
```
Comment (grantfirl) [inline ccpp/data/GFS_typedefs.meta:4866]: @JeffBeck-NOAA I think that the following should fix the GNU issue that was reported:
For each SPP variable that is conditionally allocated in GFS_typedefs.F90, (spp_prt_list, spp_stddev_cutoff, spp_var_list), please add the following metadata attribute
`active = (do_stochastically_perturbed_parameterizations)`

Comment (JeffBeck-NOAA) [reply ccpp/data/GFS_typedefs.meta:4866]: Thanks @grantfirl!  This fixed the problem!  I ended up having to apply it to spp_prt_list, spp_var_list, and spp_stddev_cutoff.

File: stochastic_physics/stochastic_physics_wrapper.F90 (hunk 1 above)
Comment (junwang-noaa) [reply stochastic_physics/stochastic_physics_wrapper.F90:112]: @JeffBeck-NOAA would you please reply to Phil?

Comment (ChunxiZhang-NOAA) [reply stochastic_physics/stochastic_physics_wrapper.F90:112]: Looks like in stochastic_physics.F90, it has been changed to 
spp_var_list_out,  spp_prt_list_out, spp_stddev_cutoff_out

**Review by ChunxiZhang-NOAA — APPROVED** [review]

File: stochastic_physics/stochastic_physics_wrapper.F90 (hunk 1 above)
Comment (JeffBeck-NOAA) [reply stochastic_physics/stochastic_physics_wrapper.F90:112]: @pjpegion, @junwang-noaa, @ChunxiZhang-NOAA, yes, this problem was fixed.

**Review by SamuelTrahanNOAA — APPROVED** [review]

**Review by junwang-noaa — APPROVED** [review]

**Comment by grantfirl** [comment]
@JeffBeck-NOAA The ccpp-physics PR has been merged. Please revert the changes to .gitmodules, checkout the top of main for ccpp-physics, 'git add' ccpp-physics, commit, and push to GitHub in preparation for the fv3atm merge.

**Comment by JeffBeck-NOAA** [comment]
@grantfirl, the fv3atm PR should be ready to go!

**Comment by BrianCurtis-NOAA** [comment]
> @grantfirl, the fv3atm PR should be ready to go!

@JeffBeck-NOAA Not quite. The branch in `.gitmodules` was not reverted to `main` (currently `bugfix/SPP`)

**Comment by BrianCurtis-NOAA** [comment]
All looks OK to me now. Merging.

**Pull request merged**